specification.md

```diff
@@ -4313,28 +4429,414 @@ transition: IssueRefundToCustomer
 
 </details>
 
-Error definitions describe checked errors that can occur during workflow execution and how to handle them.
+Error definitions are [RFC7807](https://datatracker.ietf.org/doc/html/rfc7807) compliant descriptions of errors that are produced by/originating from the execution of a workflow. Runtimes use them to describe workflow related errors in a user-friendly, technology agnostic and cross-platform way.
+
+Property `instance` identifies the component within a workflow definition from which the described error originates. It is set by runtimes when throwing an error.
+
+For example, in the above definition, the source '/states/0/actions/0' indicates that the error originates from the execution of the first action of the first state of the workflow definitions.
```

Comment:
I think instances should be `states/<stateName>/actions/<actionName>` , taking benefit of the fact that both state and action name are now mandatory

Reply:
Very good point! Thank you!

Reply:
Actually, I was trying to implement it, I realized this is not possible as is: it would result in an invalid JsonPointer.

This is because `states` is an array, and so are `actions`.

As a matter of fact:

```yaml
states:
  - name: state-1
    actions:
      - name: action-1
        functionRef: 
          refName: my-function
```

... is referenced, using JsonPointers by: `/states/0/actions/0`

However:

```yaml
states:
  state-1
    actions:
      action-1
        functionRef: 
          refName: my-function
```

... is referenced, using JsonPointers, by: `/states/state-1/actions/action-1`

Reply:
So `states` and `actions` should be refactored to maps instead?

```yaml
states
    state1: 
        actions:
            action1:
    state2:
```

So whatever it's in `states` attribute should be considered a key. I think we can consider doing this refactor. WDYT?

Reply:
> So states and actions should be refactored to maps instead?
IMHO no: it's a (unconvenient) design we would then have to apply everywhere, just for the sake of having names in JsonPointers.

Reply:
I thought it could be beneficial long-term in other aspects of the spec. Maybe I'm into JsonPointers too much :)

I guess array indexed path is fine since the states order are fixed based on the definition. 👍

Reply:
Well, if we are to do this, and I think it's the best approach (map everywhere), we should do it asap.

Could you open an issue on the matter? I'll open a PR for that.

Reply:
https://github.com/serverlessworkflow/specification/issues/831